flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -623,6 +622,7 @@ private void checkResourceRequirements() {
             if (resourceAllocator.isSupported()
                     && !taskManagerTracker.getPendingTaskManagers().isEmpty()) {
                 taskManagerTracker.replaceAllPendingAllocations(Collections.emptyMap());
+                checkTaskManagerReleasable();
```

Comment:
@huwh `checkTaskManagerReleasable()` returns a boolean value that is not verified in any way? Is this intentionally done? The same applies to line 680.

Reply:
I created FLINK-34588 to cover this issue.